src/sinks/aws_kinesis/config.rs

```diff
@@ -52,6 +52,13 @@ pub struct KinesisSinkBaseConfig {
     #[serde(default)]
     pub auth: AwsAuthentication,
 
+    /// Whether or not to retry successful requests containing partial failures.
+    ///
+    /// Note: this can cause duplicates in Firehose.
```

Comment:
🤔 should this be marked as a warning in the cue docs, to call more attention to it?

Reply:
We currently don't have a way to do this in the code with the `#[configurable]` trait but it is something we are looking into.
For now, this is an example of how to achieve that:

https://github.com/vectordotdev/vector/blob/bf7d79623c0b575dd0bb6f851cc12c15cea5eb5f/website/cue/reference/components/sources/http_client.cue#L55-L57

I think we could do that for the firehose sink, and replace that doc comment on line 57 with just a regular comment, so it doesn't show up in the auto generated docs.

Reply:
I already refactored it. But I am not sure it is what you want. If it is not what you want, could you help to push commit to https://github.com/dengmingtong/vector/tree/kinesis-retry to refactor this comment?
I have grant you to access this https://github.com/dengmingtong/vector/tree/kinesis-retry branch.

Reply:
Refactored the code according to your comments, but got build error as below:
https://app.netlify.com/sites/vector-project/deploys/64858a564a61b900081e1a4e

I don't have a good understanding of the cue document, and we need this PR merged into master urgently. If possible, could you please directly modify your comment in my repo https://github.com/dengmingtong/vector/tree/kinesis-retry. Alternatively, please give me some detailed hints on how to refactor code to meet your comments..

Reply:
https://github.com/dengmingtong/vector/compare/kinesis-retry...vectordotdev:vector:kinesis-retry

This seems to work!

Reply:
@spencergilbert 
Thank you very much, I have push a commit to fix it.
I already grant you to access https://github.com/dengmingtong/vector/tree/kinesis-retry, if there still need any fix, could you help to fix it and push the commit to https://github.com/dengmingtong/vector/tree/kinesis-retry directly..